tools/aws_lab_setup/roles/student_users/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+# create users if not defined, using integer student_total
```

Comment:
Wait. What is going on here? I'm confused. You're generating a var file and then having Ansible read it back in?

Reply:
this was made to be backwards compatible with the student vars list in the existing deployment model, it just generates the list.

Reply:
Understood on compatibility. The current provisioner requires this file exist already though. We have a separate tool for generating an inventory like this in this PR. Some portion of facilitators are likely to want to create something different or more specific whether manually or some sort of export from a registration system. So while it's an idea I'd rather not add to all of this already sizable PR with something that is redundant, optional and of limited utility. I suggest removal and docs mentioning the shell script utility that can be run as a pre-task to provisioning.

Reply:
> The current provisioner requires this file exist already though. We have a separate tool for generating an inventory like this in this PR.
If you specify a students list, this entire role gets skipped, the role is only 3 tasks and would deprecate the shell script and be easier for consumption.

Reply:
Then pull this from the PR and let's debate that separately. This is one of those things that can be pretty easily broken off to cut down on all of the action in this one.

Reply:
its a huge amount of work for me to pull that role because its called from both provision_lab.yml and teardown_lab.yml and the way I did the PR cleanly like this, if you keep requesting changes I can't PR back to network-automation anymore, I have to manually do diffs... this has been taking me about an hour today b/c we already removed the s3 website thing.  

I would rather move forward and just deprecate the non-auto generated users, it will greatly clean up everything

honestly would also love to deprecate the sendmail stuff as well as no one seems to be using it.

Reply:
This is why you don't turn in such a massive sprawling PR like this. 

I think it was a totally unnecessary addition and shouldn't have been in here without discussion with the rest of us. I'm pretty tired of this though and we already know we will need to redo everything here so whatever leave it in. This is something I will look to rip out as soon as I can make it happen.

Reply:
I on the other hand think this is a completely acceptable compromise.  It only generates the list if you don't have one defined already.  Using a shell script to generate a list of random users is asinine when you can dynamically do it in the script.  Granted in the SA fork, we just skip the file entirely as we find it completely unnecessary and just use the variable.

People that want to do something different can just create the file with what they want.  The rest of us can just have it auto-generate the file for us.  Its either this, or we add a task that uses the command module to run the shell script, but the whole point of automation is to get away from running shell scripts in the first place.

Reply:
It didn't need to be in this PR and should have been an optional separate role. Don't get me started on asinine.